packages/next/taskfile.js

```diff
@@ -53,6 +134,111 @@ export async function ncc_acorn(task, opts) {
     .target('compiled/acorn')
 }
 
+// eslint-disable-next-line camelcase
+export async function ncc_next__react_dev_overlay(task, opts) {
+  const overlayExternals = {
+    ...externals,
+    react: 'react',
+    'react-dom': 'react-dom',
+  }
+  // dev-overlay needs a newer source-map version
+  delete overlayExternals['source-map']
+
+  await task
+    .source(
+      opts.src ||
+        relative(
+          __dirname,
+          require.resolve('@next/react-dev-overlay/lib/middleware')
+        )
+    )
+    .ncc({
+      packageName: '@next/react-dev-overlay',
+      externals: overlayExternals,
+      target: 'es5',
+    })
+    .target('compiled/@next/react-dev-overlay')
+
+  await task
+    .source(
+      opts.src ||
+        relative(
+          __dirname,
+          require.resolve('@next/react-dev-overlay/lib/client')
+        )
+    )
+    .ncc({
+      packageName: '@next/react-dev-overlay',
+      externals: overlayExternals,
+      target: 'es5',
+    })
+    .target('compiled/@next/react-dev-overlay')
+
+  const clientFile = join(
+    __dirname,
+    'compiled/@next/react-dev-overlay/client.js'
+  )
+  const content = fs.readFileSync(clientFile, 'utf8')
+  // remove AMD define branch as this forces the module to not
+  // be treated as commonjs in serverless mode
+  fs.writeFileSync(
+    clientFile,
+    content.replace(
+      'if(typeof define=="function"&&typeof define.amd=="object"&&define.amd){r.platform=v;define((function(){return v}))}else ',
+      ''
+    )
+  )
+}
+
+// eslint-disable-next-line camelcase
+externals['watchpack'] = 'next/dist/compiled/watchpack'
+export async function ncc_watchpack(task, opts) {
+  await task
+    .source(opts.src || relative(__dirname, require.resolve('watchpack')))
+    .ncc({ packageName: 'watchpack', externals })
+    .target('compiled/watchpack')
+}
+
+// eslint-disable-next-line camelcase
+externals['chalk'] = 'next/dist/compiled/chalk'
+export async function ncc_chalk(task, opts) {
+  await task
+    .source(opts.src || relative(__dirname, require.resolve('chalk')))
+    .ncc({ packageName: 'chalk', externals })
+    .target('compiled/chalk')
+
+  const content = fs.readFileSync(
+    join(__dirname, 'compiled/chalk/index.js'),
+    'utf8'
+  )
+  // ensure undefined process.argv is handled for web runtime
+  // TODO: should chalk be being included in web runtime?
```

Comment:
By "web" runtime, you mean the current middleware runtime? I would think no, chalk shouldn't be included.

Reply:
Yeah seems it's being pulled in with the `react-18` and `react-streaming-and-server-components` test suites cc @huozhi

Reply:
I think we have excluded it in webpack config.

```js
...(webServerRuntime
        ? [{ etag: '{}', chalk: '{}', 'react-dom': '{}' }]
```

Reply:
Ah ok looks like we just need to update that to the compiled location now, will open a follow-up PR that includes that

Reply:
Updated in https://github.com/vercel/next.js/pull/32717, looks like `etag` needed to be updated as well